### ISSUE TITLE
Proposals: Implement rules of voice decay and rollover

### ISSUE THREAD

**Issue by n13** [JoinSEEDS/seeds-smart-contracts#15]
## Goals of voice decay and rollover

- Incentivize users to participate -> decay
- Dis-incentivize users to use their vote on things they don't care about -> rollover
- Prevent hoarding of votes -> time limit on rollover

## Potential better solutions
We might be able to come up with simpler rules to reach the same goals - open to suggestions.

## Solution

Voice starts decaying halfway through a cycle
- If a user has not used their voice on anything, it starts linearly declining towards 0 at the end of the cycle
- if a user uses their voice or votes indifferent, then their voice does not decay
- if a user used their voice during the cycle it carries over to the next cycle
- max voice anyone could accumulate is 3 cycles worth - this timespan is in settings
- In order to implement this we need to have a way to store the unused voice for each user each cycle. 

## Implementation suggestion

I would suggest having a rollover table scoped by user name - on each new cycle, new voice for that cycle is calculated and added to the rollover table with timestamp

Then we calculate total voice of all rollover timestamps (which includes the latest) < 3 cycles old and add it to the voice table (so the frontend can show total voice, and everyone knows what total voice is)

While we iterate we also delete entries > 3 cycles old

To implement voice decay, both the voice table entry and all rollover entries are decayed at the same time and same rate. Note that decay only starts 1/2 way through the cycle.

**Comment by IanMendozaJaimes** [comment]
Hello, I have some questions:
- If voice starts decaying and then a user uses their voice, the voice stops decaying, right?
- What do you mean by vote indifferent?
- How could I know in what part of the cycle we are, I mean, how could I know when the 1/2 cycle is reached?
- What would be the period of time the voicedecay function would be executed?

**Comment by Rieki7** [comment]
Voice=trust
1. Yes once votes are cast the "current trust" is moved to "saved trust" and the decay stops (as decay is only affecting "current trust"
2. Indifferent being - abstain. Its a "vote" but one where no trust credits are cast. This is just to signify - I came, I saw, I'm indifferent. 

3. Each cycles starts at full and new moon. Half cycles are at quarter and 3/4 moons. 

4. Trust decay starts at half cycle and finishes at start of new cycle where the Full Trust is diminished within that week.
